[WIP] Add minimal tests.

beechnut opened the pull request:
Minimal unit tests are passing.

Minimal integration tests are not passing, because of an issue related to a missing database table for another 'app'.

Working toward #287 and #320.

Commit:
[WIP] Minimal tests.

Minimal unit tests are passing.
Minimal integration tests are not passing, issue related
  to a database table for another app.

neilmb commented on nrm_django/grants/tests/models/test_category.py:
```diff
@@ -0,0 +1,9 @@
+from grants.models import Category
+
+class CategoryTestCase(TestCase):
+    def test_something(self):
```
maybe `test_creation`?

neilmb commented on nrm_django/grants/tests/models/test_grant.py:
```diff
@@ -0,0 +1,10 @@
+from grants.models import Grant
+
+class GrantTestCase(TestCase):
+    def test_something(self):
```
Same question about naming

neilmb commented on nrm_django/grants/tests/models/test_grant_authority.py:
```diff
@@ -0,0 +1,9 @@
+from grants.models import GrantAuthority
+
+class GrantAuthorityTestCase(TestCase):
+    def test_something(self):
```
Same.

Commit:
Improve database in service of tests; fix first integration test

Integration tests were raising two major issues:

1. Test tables were missing because of `managed=False` settings in migrations & models.
2. SQL `DISTINCT` clauses were touched, but the sqlite backend doesn't support that syntax.

So, this commit:

- Comments out all `managed=False` settings. (Is this the right workflow for us?)
- Generates migrations to update all the models that were previously unmanaged.
- Updates settings so the default backend is PostgreSQL and will work with the Docker Compose environment.
- Updates urls.py to use the improved grants/change template for projects with an X in the ID.

Commit:
Merge branch 'development' into 287-unit-tests

Commit:
Update test names and improve code style via flake8

Commit:
Setup Docker container for Postgres in CI, and reformat code with Black

Commit:
Merge branch 'development' into 287-unit-tests
